### ISSUE TITLE
0.44.0 Fails to run on heroku

### ISSUE THREAD

**Issue by glmvrml** [metabase/metabase-buildpack#18]
```
2022-08-04T20:00:57.328080+00:00 app[web.1]: Exception in thread "main" java.lang.Error: Circular loading of installed providers detected
2022-08-04T20:00:57.328357+00:00 app[web.1]: at java.nio.file.spi.FileSystemProvider.installedProviders(FileSystemProvider.java:161)
2022-08-04T20:00:57.328384+00:00 app[web.1]: at java.nio.file.Paths.get(Paths.java:141)
2022-08-04T20:00:57.328408+00:00 app[web.1]: at org.apache.logging.log4j.core.util.Source.toFile(Source.java:55)
2022-08-04T20:00:57.328436+00:00 app[web.1]: at org.apache.logging.log4j.core.util.Source.<init>(Source.java:142)
2022-08-04T20:00:57.328462+00:00 app[web.1]: at org.apache.logging.log4j.core.config.ConfigurationSource.<init>(ConfigurationSource.java:139)
2022-08-04T20:00:57.328486+00:00 app[web.1]: at org.apache.logging.log4j.core.config.ConfigurationSource.getConfigurationSource(ConfigurationSource.java:388)
2022-08-04T20:00:57.328520+00:00 app[web.1]: at org.apache.logging.log4j.core.config.ConfigurationSource.fromResource(ConfigurationSource.java:366)
2022-08-04T20:00:57.328547+00:00 app[web.1]: at org.apache.logging.log4j.core.config.ConfigurationFactory$Factory.getConfiguration(ConfigurationFactory.java:554)
2022-08-04T20:00:57.328593+00:00 app[web.1]: at org.apache.logging.log4j.core.config.ConfigurationFactory$Factory.getConfiguration(ConfigurationFactory.java:483)
2022-08-04T20:00:57.328603+00:00 app[web.1]: at org.apache.logging.log4j.core.config.ConfigurationFactory.getConfiguration(ConfigurationFactory.java:325)
2022-08-04T20:00:57.328628+00:00 app[web.1]: at org.apache.logging.log4j.core.LoggerContext.reconfigure(LoggerContext.java:690)
2022-08-04T20:00:57.328652+00:00 app[web.1]: at org.apache.logging.log4j.core.LoggerContext.reconfigure(LoggerContext.java:711)
2022-08-04T20:00:57.328681+00:00 app[web.1]: at org.apache.logging.log4j.core.LoggerContext.start(LoggerContext.java:253)
2022-08-04T20:00:57.328705+00:00 app[web.1]: at org.apache.logging.log4j.core.impl.Log4jContextFactory.getContext(Log4jContextFactory.java:155)
2022-08-04T20:00:57.328729+00:00 app[web.1]: at org.apache.logging.log4j.core.impl.Log4jContextFactory.getContext(Log4jContextFactory.java:47)
2022-08-04T20:00:57.328757+00:00 app[web.1]: at org.apache.logging.log4j.LogManager.getContext(LogManager.java:196)
2022-08-04T20:00:57.328781+00:00 app[web.1]: at org.apache.logging.log4j.spi.AbstractLoggerAdapter.getContext(AbstractLoggerAdapter.java:137)
2022-08-04T20:00:57.328806+00:00 app[web.1]: at org.apache.logging.slf4j.Log4jLoggerFactory.getContext(Log4jLoggerFactory.java:55)
2022-08-04T20:00:57.328837+00:00 app[web.1]: at org.apache.logging.log4j.spi.AbstractLoggerAdapter.getLogger(AbstractLoggerAdapter.java:47)
2022-08-04T20:00:57.328861+00:00 app[web.1]: at org.apache.logging.slf4j.Log4jLoggerFactory.getLogger(Log4jLoggerFactory.java:33)
2022-08-04T20:00:57.328884+00:00 app[web.1]: at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:363)
2022-08-04T20:00:57.328911+00:00 app[web.1]: at com.mchange.v2.log.slf4j.Slf4jMLog.getMLogger(Slf4jMLog.java:75)
2022-08-04T20:00:57.328943+00:00 app[web.1]: at com.mchange.v2.log.MLog.getMLogger(MLog.java:398)
2022-08-04T20:00:57.328966+00:00 app[web.1]: at com.mchange.v2.log.MLog.getLogger(MLog.java:281)
2022-08-04T20:00:57.328993+00:00 app[web.1]: at com.mchange.v2.log.MLog.refreshConfig(MLog.java:157)
2022-08-04T20:00:57.329014+00:00 app[web.1]: at com.mchange.v2.log.MLog.<clinit>(MLog.java:52)
2022-08-04T20:00:57.329037+00:00 app[web.1]: at com.mchange.v2.naming.JavaBeanReferenceMaker.<clinit>(JavaBeanReferenceMaker.java:51)
2022-08-04T20:00:57.329068+00:00 app[web.1]: at com.mchange.v2.c3p0.impl.PoolBackedDataSourceBase.<clinit>(PoolBackedDataSourceBase.java:263)
2022-08-04T20:00:57.329091+00:00 app[web.1]: at java.lang.Class.forName0(Native Method)
2022-08-04T20:00:57.329113+00:00 app[web.1]: at java.lang.Class.forName(Class.java:348)
2022-08-04T20:00:57.329142+00:00 app[web.1]: at clojure.lang.RT.classForName(RT.java:2209)
2022-08-04T20:00:57.329163+00:00 app[web.1]: at clojure.lang.RT.classForName(RT.java:2218)
2022-08-04T20:00:57.329187+00:00 app[web.1]: at metabase.db.connection_pool_setup$fn__20052.<clinit>(connection_pool_setup.clj:20)
2022-08-04T20:00:57.329215+00:00 app[web.1]: at metabase.db.connection_pool_setup__init.load(Unknown Source)
2022-08-04T20:00:57.329237+00:00 app[web.1]: at metabase.db.connection_pool_setup__init.<clinit>(Unknown Source)
2022-08-04T20:00:57.329258+00:00 app[web.1]: at java.lang.Class.forName0(Native Method)
2022-08-04T20:00:57.329284+00:00 app[web.1]: at java.lang.Class.forName(Class.java:348)
2022-08-04T20:00:57.329307+00:00 app[web.1]: at clojure.lang.RT.classForName(RT.java:2209)
2022-08-04T20:00:57.329329+00:00 app[web.1]: at clojure.lang.RT.classForName(RT.java:2218)
2022-08-04T20:00:57.329355+00:00 app[web.1]: at clojure.lang.RT.loadClassForName(RT.java:2237)
2022-08-04T20:00:57.329379+00:00 app[web.1]: at clojure.lang.RT.load(RT.java:449)
2022-08-04T20:00:57.329400+00:00 app[web.1]: at clojure.lang.RT.load(RT.java:424)
2022-08-04T20:00:57.329422+00:00 app[web.1]: at clojure.core$load$fn__6908.invoke(core.clj:6161)
2022-08-04T20:00:57.329449+00:00 app[web.1]: at clojure.core$load.invokeStatic(core.clj:6160)
2022-08-04T20:00:57.329473+00:00 app[web.1]: at clojure.core$load.doInvoke(core.clj:6144)
2022-08-04T20:00:57.329495+00:00 app[web.1]: at clojure.lang.RestFn.invoke(RestFn.java:408)
2022-08-04T20:00:57.329521+00:00 app[web.1]: at clojure.core$load_one.invokeStatic(core.clj:5933)
2022-08-04T20:00:57.329543+00:00 app[web.1]: at clojure.core$load_one.invoke(core.clj:5928)
2022-08-04T20:00:57.329566+00:00 app[web.1]: at clojure.core$load_lib$fn__6850.invoke(core.clj:5975)
2022-08-04T20:00:57.329591+00:00 app[web.1]: at clojure.core$load_lib.invokeStatic(core.clj:5974)
2022-08-04T20:00:57.329617+00:00 app[web.1]: at clojure.core$load_lib.doInvoke(core.clj:5953)
2022-08-04T20:00:57.329638+00:00 app[web.1]: at clojure.lang.RestFn.applyTo(RestFn.java:142)
2022-08-04T20:00:57.329663+00:00 app[web.1]: at clojure.core$apply.invokeStatic(core.clj:669)
2022-08-04T20:00:57.329687+00:00 app[web.1]: at clojure.core$load_libs.invokeStatic(core.clj:6016)
2022-08-04T20:00:57.329709+00:00 app[web.1]: at clojure.core$load_libs.doInvoke(core.clj:6000)
2022-08-04T20:00:57.329731+00:00 app[web.1]: at clojure.lang.RestFn.applyTo(RestFn.java:137)
2022-08-04T20:00:57.329757+00:00 app[web.1]: at clojure.core$apply.invokeStatic(core.clj:669)
2022-08-04T20:00:57.329779+00:00 app[web.1]: at clojure.core$require.invokeStatic(core.clj:6038)
2022-08-04T20:00:57.329802+00:00 app[web.1]: at clojure.core$require.doInvoke(core.clj:6038)
2022-08-04T20:00:57.329828+00:00 app[web.1]: at clojure.lang.RestFn.invoke(RestFn.java:436)
2022-08-04T20:00:57.329855+00:00 app[web.1]: at metabase.db.connection$loading__6789__auto____20106.invoke(connection.clj:1)
2022-08-04T20:00:57.329877+00:00 app[web.1]: at metabase.db.connection__init.load(Unknown Source)
2022-08-04T20:00:57.329907+00:00 app[web.1]: at metabase.db.connection__init.<clinit>(Unknown Source)
2022-08-04T20:00:57.329929+00:00 app[web.1]: at java.lang.Class.forName0(Native Method)
2022-08-04T20:00:57.329953+00:00 app[web.1]: at java.lang.Class.forName(Class.java:348)
2022-08-04T20:00:57.329982+00:00 app[web.1]: at clojure.lang.RT.classForName(RT.java:2209)
2022-08-04T20:00:57.330006+00:00 app[web.1]: at clojure.lang.RT.classForName(RT.java:2218)
2022-08-04T20:00:57.330028+00:00 app[web.1]: at clojure.lang.RT.loadClassForName(RT.java:2237)
2022-08-04T20:00:57.330054+00:00 app[web.1]: at clojure.lang.RT.load(RT.java:449)
2022-08-04T20:00:57.330075+00:00 app[web.1]: at clojure.lang.RT.load(RT.java:424)
2022-08-04T20:00:57.330097+00:00 app[web.1]: at clojure.core$load$fn__6908.invoke(core.clj:6161)
2022-08-04T20:00:57.330122+00:00 app[web.1]: at clojure.core$load.invokeStatic(core.clj:6160)
2022-08-04T20:00:57.330144+00:00 app[web.1]: at clojure.core$load.doInvoke(core.clj:6144)
2022-08-04T20:00:57.330167+00:00 app[web.1]: at clojure.lang.RestFn.invoke(RestFn.java:408)
2022-08-04T20:00:57.330190+00:00 app[web.1]: at clojure.core$load_one.invokeStatic(core.clj:5933)
2022-08-04T20:00:57.330214+00:00 app[web.1]: at clojure.core$load_one.invoke(core.clj:5928)
2022-08-04T20:00:57.330236+00:00 app[web.1]: at clojure.core$load_lib$fn__6850.invoke(core.clj:5975)
2022-08-04T20:00:57.330259+00:00 app[web.1]: at clojure.core$load_lib.invokeStatic(core.clj:5974)
2022-08-04T20:00:57.330285+00:00 app[web.1]: at clojure.core$load_lib.doInvoke(core.clj:5953)
2022-08-04T20:00:57.330307+00:00 app[web.1]: at clojure.lang.RestFn.applyTo(RestFn.java:142)
2022-08-04T20:00:57.330328+00:00 app[web.1]: at clojure.core$apply.invokeStatic(core.clj:669)
2022-08-04T20:00:57.330354+00:00 app[web.1]: at clojure.core$load_libs.invokeStatic(core.clj:6016)
2022-08-04T20:00:57.330377+00:00 app[web.1]: at clojure.core$load_libs.doInvoke(core.clj:6000)
2022-08-04T20:00:57.330398+00:00 app[web.1]: at clojure.lang.RestFn.applyTo(RestFn.java:137)
2022-08-04T20:00:57.330422+00:00 app[web.1]: at clojure.core$apply.invokeStatic(core.clj:669)
2022-08-04T20:00:57.330444+00:00 app[web.1]: at clojure.core$require.invokeStatic(core.clj:6038)
2022-08-04T20:00:57.330466+00:00 app[web.1]: at clojure.core$require.doInvoke(core.clj:6038)
2022-08-04T20:00:57.330488+00:00 app[web.1]: at clojure.lang.RestFn.invoke(RestFn.java:457)
2022-08-04T20:00:57.330515+00:00 app[web.1]: at metabase.db$loading__6789__auto____35568.invoke(db.clj:1)
2022-08-04T20:00:57.330536+00:00 app[web.1]: at metabase.db__init.load(Unknown Source)
2022-08-04T20:00:57.330565+00:00 app[web.1]: at metabase.db__init.<clinit>(Unknown Source)
2022-08-04T20:00:57.330609+00:00 app[web.1]: at java.lang.Class.forName0(Native Method)
2022-08-04T20:00:57.330633+00:00 app[web.1]: at java.lang.Class.forName(Class.java:348)
2022-08-04T20:00:57.330659+00:00 app[web.1]: at clojure.lang.RT.classForName(RT.java:2209)
2022-08-04T20:00:57.330778+00:00 app[web.1]: at clojure.lang.RT.classForName(RT.java:2218)
2022-08-04T20:00:57.330804+00:00 app[web.1]: at clojure.lang.RT.loadClassForName(RT.java:2237)
2022-08-04T20:00:57.330832+00:00 app[web.1]: at clojure.lang.RT.load(RT.java:449)
2022-08-04T20:00:57.330855+00:00 app[web.1]: at clojure.lang.RT.load(RT.java:424)
2022-08-04T20:00:57.330877+00:00 app[web.1]: at clojure.core$load$fn__6908.invoke(core.clj:6161)
2022-08-04T20:00:57.330903+00:00 app[web.1]: at clojure.core$load.invokeStatic(core.clj:6160)
2022-08-04T20:00:57.330926+00:00 app[web.1]: at clojure.core$load.doInvoke(core.clj:6144)
2022-08-04T20:00:57.330948+00:00 app[web.1]: at clojure.lang.RestFn.invoke(RestFn.java:408)
2022-08-04T20:00:57.330973+00:00 app[web.1]: at clojure.core$load_one.invokeStatic(core.clj:5933)
2022-08-04T20:00:57.330997+00:00 app[web.1]: at clojure.core$load_one.invoke(core.clj:5928)
2022-08-04T20:00:57.331024+00:00 app[web.1]: at clojure.core$load_lib$fn__6850.invoke(core.clj:5975)
2022-08-04T20:00:57.331041+00:00 app[web.1]: at clojure.core$load_lib.invokeStatic(core.clj:5974)
2022-08-04T20:00:57.331064+00:00 app[web.1]: at clojure.core$load_lib.doInvoke(core.clj:5953)
2022-08-04T20:00:57.331088+00:00 app[web.1]: at clojure.lang.RestFn.applyTo(RestFn.java:142)
2022-08-04T20:00:57.331112+00:00 app[web.1]: at clojure.core$apply.invokeStatic(core.clj:669)
2022-08-04T20:00:57.331139+00:00 app[web.1]: at clojure.core$load_libs.invokeStatic(core.clj:6016)
2022-08-04T20:00:57.331163+00:00 app[web.1]: at clojure.core$load_libs.doInvoke(core.clj:6000)
2022-08-04T20:00:57.331185+00:00 app[web.1]: at clojure.lang.RestFn.applyTo(RestFn.java:137)
2022-08-04T20:00:57.331213+00:00 app[web.1]: at clojure.core$apply.invokeStatic(core.clj:669)
2022-08-04T20:00:57.331236+00:00 app[web.1]: at clojure.core$require.invokeStatic(core.clj:6038)
2022-08-04T20:00:57.331262+00:00 app[web.1]: at clojure.core$require.doInvoke(core.clj:6038)
2022-08-04T20:00:57.331290+00:00 app[web.1]: at clojure.lang.RestFn.invoke(RestFn.java:3894)
2022-08-04T20:00:57.331309+00:00 app[web.1]: at metabase.core$loading__6789__auto____83702.invoke(core.clj:1)
2022-08-04T20:00:57.331326+00:00 app[web.1]: at metabase.core__init.load(Unknown Source)
2022-08-04T20:00:57.331344+00:00 app[web.1]: at metabase.core__init.<clinit>(Unknown Source)
2022-08-04T20:00:57.331374+00:00 app[web.1]: at java.lang.Class.forName0(Native Method)
2022-08-04T20:00:57.331388+00:00 app[web.1]: at java.lang.Class.forName(Class.java:348)
2022-08-04T20:00:57.331407+00:00 app[web.1]: at clojure.lang.RT.classForName(RT.java:2209)
2022-08-04T20:00:57.331437+00:00 app[web.1]: at clojure.lang.RT.classForName(RT.java:2218)
2022-08-04T20:00:57.331453+00:00 app[web.1]: at clojure.lang.RT.loadClassForName(RT.java:2237)
2022-08-04T20:00:57.331485+00:00 app[web.1]: at clojure.lang.RT.load(RT.java:449)
2022-08-04T20:00:57.331507+00:00 app[web.1]: at clojure.lang.RT.load(RT.java:424)
2022-08-04T20:00:57.331548+00:00 app[web.1]: at clojure.core$load$fn__6908.invoke(core.clj:6161)
2022-08-04T20:00:57.331565+00:00 app[web.1]: at clojure.core$load.invokeStatic(core.clj:6160)
2022-08-04T20:00:57.331598+00:00 app[web.1]: at clojure.core$load.doInvoke(core.clj:6144)
2022-08-04T20:00:57.331614+00:00 app[web.1]: at clojure.lang.RestFn.invoke(RestFn.java:408)
2022-08-04T20:00:57.331633+00:00 app[web.1]: at clojure.lang.Var.invoke(Var.java:384)
2022-08-04T20:00:57.331659+00:00 app[web.1]: at clojure.lang.Util.loadWithClass(Util.java:251)
2022-08-04T20:00:57.331677+00:00 app[web.1]: at metabase.core.<clinit>(Unknown Source)
2022-08-04T20:00:57.471359+00:00 heroku[web.1]: Process exited with status 1
2022-08-04T20:00:57.527129+00:00 heroku[web.1]: State changed from starting to crashed
```

**Comment by coreyaus** [comment]
I had this same issue and was able to resolve it by following a couple of the suggestions from an open pull request in the `metabase-deploy` repository

TL;DR - following these suggestions worked for me with the Heroku-22 stack: https://github.com/metabase/metabase-deploy/pull/28

I had previously tried the change visible here: https://github.com/metabase/metabase-deploy/pull/30 but that didn't work for me on the Heroku-22 stack - maybe I did something wrong so that PR may work for others, but for an immediate patch the older PR 28 did the trick for me 😄

**Comment by coreyaus** [comment]
PS: Also see https://github.com/metabase/metabase/issues/24619

**Comment by paoliniluis** [comment]
Fixes were done in the deploy repo. Closing 🙂
